src/types/types.ts

```diff
@@ -14,14 +19,21 @@ export type CompareResultsTableFilterableCell = {
   ) => boolean;
 };
 
+type SortableCell = {
+  name: string;
+  sortFunction: (
+    resultA: CompareResultsItem,
+    resultB: CompareResultsItem,
+  ) => number;
+};
+
+export type CompareResultsTableFilterableCell = BasicCell & FilterableCell;
+
 export type CompareResultsTableCell =
-  | {
-      name?: string;
-      filter?: false;
-      key: string;
-      gridWidth?: string;
-    }
-  | CompareResultsTableFilterableCell;
+  | BasicCell
+  | (BasicCell & SortableCell)
+  | CompareResultsTableFilterableCell
+  | (CompareResultsTableFilterableCell & SortableCell);
```

Comment:
I thought that splitting the types into subtypes would made it more readable

Reply:
Thanks for this, but maybe more documentation is needed later to understand how we're successfully capturing all the possible type cases for `CompareResultsTableCell` so it doesn't produce an error? Is it standard practice to have all these possible options listed like this for a type or would it be too much to find the type through a function? Just wondering.

Reply:
It's clearly not ideal especially if there are even more combinations later. I thought this could be a problem for later _if_ more types are indeed needed :-) Definitely I can add some comments to explain the types, I should have done that before the review indeed!

Also I could do something such as:
```ts
export type CompareResultsTableFilterableCell = BasicCell & FilterableCell;
type MaybeSortable<C extends BasicCell> = C | (C & SortableCell);
export type CompareResultsTableCell =
  | MaybeSortable<BasicCell>
  | MaybeSortable<CompareResultsTableFilterableCell>;
```
to factor the union with `SortableCell` into a separate type. Would that be better?


I'm curious what you have in mind when you write `find the type through a function`. Can you provide an example (even a simpler one)?

Reply:
The `MaybeSortable` utility type is more aligned with what I was thinking when I said `find the type through a function`, which would require more effort and be more confusing. 
Example: 

```
//you'd have to disable the any tsc eslint error
const findType = (cell: any} => {
  if ('filter' in cell) {
    return 'filterable';
  }
  if ('sortFunction' in cell) {
    return 'sortable';
  }
  return 'basic';
};

// pass down findType
//set state and the appropriate types
// do things based on findType
// more work and have to figure out other things :(
```

-- also I think my potential type error concerns are already handled through differentiating between the cell types so it would be redundant [TableHeader.tsx#L376](https://github.com/mozilla/perfcompare/blob/df5efb51832d0dbe1492b8b63c24b1136f9287bf/src/components/CompareResults/TableHeader.tsx#L376)